### PR TITLE
18 Add pay-to address error checking / validation

### PR DESCRIPTION
**Pull request by paOol**
https://github.com/PayButton/paybutton/issues/18

### CODE REVIEW

File: react/src/components/PayButton/PayButton.tsx [hunk 1]
```diff
@@ -55,6 +56,16 @@ export const PayButton = (props: PayButtonProps): React.ReactElement => {
     onSuccess?.(txid, amount);
   };
 
+  useEffect(() => {
+    if (validateCashAddress(to)) {
+      setDisabled(false);
+      setErrorMsg('');
+    } else {
+      setDisabled(true);
+      setErrorMsg('Invalid Recipient');
```
[the remark this answers is not in the document]
Comment (Klakurka) [reply react/src/components/PayButton/PayButton.tsx:65]: Let's also add a case for `to` being missing or blank. That will be an obvious one as well.

`Missing Recipient` would be fine for that message.

Double check that it's working on the HTML side as well.

**Review by Klakurka — CHANGES_REQUESTED** [review]
On build, there are new errors/warnings: 
image

File: react/src/components/PayButton/PayButton.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (paOol) [reply react/src/components/PayButton/PayButton.tsx:65]: after deleting node_modules and package-lock.json, I was able to npm i and build without errors. 

also added the other changes.

**Comment by Klakurka** [comment]
This looks like still an issue during the build process (for paybutton.js):

image

File: react/src/components/PayButton/PayButton.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Klakurka) [reply react/src/components/PayButton/PayButton.tsx:65]: Is it too much of a pain to have the "missing recipient" behaviour be the same even if `to=` is actually missing? Right now the button just doesn't display and shows a console error. The error is helpful, but only for people who know how to open up the console.

**Comment by Klakurka** [comment]
After deleting all node_modules files:

image

It might be different if the entire checked out folder is deleted and remade, not sure (haven't tried).

**Comment by Klakurka** [comment]
The warning doesn't seem to cause any issues and isn't happening on Paul's env so we'll just leave it for now.

**Pull request merged**